src/network/networkd-ndisc.c

```diff
@@ -207,7 +209,20 @@ static void ndisc_router_process_autonomous_prefix(Link *link, sd_ndisc_router *
         address->prefixlen = prefixlen;
         address->flags = IFA_F_NOPREFIXROUTE|IFA_F_MANAGETEMPADDR;
         address->cinfo.ifa_prefered = lifetime_preferred;
-        address->cinfo.ifa_valid = lifetime_valid;
+
+        /* see RFC4862 section 5.5.3.e */
+        r = address_get(link, address->family, &address->in_addr, address->prefixlen, &existing_address);
+        if(r > 0){
+            if(lifetime_valid > NDISC_PREFIX_LFT_MIN || lifetime_valid > existing_address->cinfo.ifa_valid)
+                address->cinfo.ifa_valid = lifetime_valid;
+            else if(existing_address->cinfo.ifa_valid <= NDISC_PREFIX_LFT_MIN)
+                address->cinfo.ifa_valid = existing_address->cinfo.ifa_valid;
```

Comment:
ACK. We don't have secured Router Advertisments so the bottom half of section 5.5.3, part e), 2nd bullet does not apply.